src/bin/languageserver/mod.rs

```diff
@@ -1990,24 +1998,25 @@ impl LanguageServer for SolangServer {
         if let Ok(path) = uri.to_file_path() {
             let files = &self.files.lock().await;
             if let Some(cache) = files.caches.get(&path) {
-                let offset = cache
+                if let Some(offset) = cache
                     .file
-                    .get_offset(pos.line as usize, pos.character as usize);
-
-                // The shortest hover for the position will be most informative
-                if let Some(hover) = cache
-                    .hovers
-                    .find(offset, offset + 1)
-                    .min_by(|a, b| (a.stop - a.start).cmp(&(b.stop - b.start)))
+                    .get_offset(pos.line as usize, pos.character as usize)
                 {
-                    let range = get_range_exclusive(hover.start, hover.stop, &cache.file);
-
-                    return Ok(Some(Hover {
-                        contents: HoverContents::Scalar(MarkedString::from_markdown(
-                            hover.val.to_string(),
-                        )),
-                        range: Some(range),
-                    }));
+                    // The shortest hover for the position will be most informative
+                    if let Some(hover) = cache
+                        .hovers
+                        .find(offset, offset + 1)
+                        .min_by(|a, b| (a.stop - a.start).cmp(&(b.stop - b.start)))
+                    {
+                        let range = get_range_exclusive(hover.start, hover.stop, &cache.file);
+
+                        return Ok(Some(Hover {
+                            contents: HoverContents::Scalar(MarkedString::from_markdown(
+                                hover.val.to_string(),
+                            )),
+                            range: Some(range),
+                        }));
+                    }
```

Comment:
The `Ok(None)` code path is never tested (This is why we have this alert from the bot).

Reply:
Hello @LucasSte ! 
Does the bot only consider tests written in rust? Because currently we test the language server code primarily through tests written in typescript.

Reply:
The bot considers all code paths. Ideally we would have a vscode test which checks this path.

Reply:
This is tested now.